### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by danmanj**
These are both cosmetic changes and changes that are needed for homa to work with distbench, our open-source distributed system benchmark.

### CODE REVIEW

**Comment by johnousterhout** [comment]
Thanks for the pull request. I'm about to go off the grid for a couple of weeks of vacation, so it will probably be early September before I can look at these changes in detail. However, I'm very interested in incorporating them and will definitely work on them when I get back.

**Comment by danmanj** [comment]
I see you are back from vacation, and there are some merge conflicts with the latest code. We really should chat ASAP, as the merge conflicts might get out of hand otherwise.

**Comment by johnousterhout** [comment]
I'd be happy to chat. In particular, I'd love to learn more about what you
are doing with Homa and how it has been working for you so far.

Sorry about the merge conflicts. These commits have been sitting in my repo
for a while, but I didn't want to check them in immediately before leaving
on vacation, so I did it this week when I got back.

It's probably going to be another week or so before I can get to your pull
request: I have some course material preparation that is up against a
deadline, so I have to do that first (honestly, I'd much rather be working
on your pull request!).

How about suggesting a few times that work for you for a virtual chat? My
schedule is pretty flexible for the next couple of weeks.

-John-

**Comment by johnousterhout** [comment]
I'm about to start merging this pull request, so please don't add more commits to it (start a fresh pull request if needed).

-John-

**Comment by danmanj** [comment]
I was about to rebase to resolve the merge conflicts at my end, let me know if you want that.

File: homa_socktab.c [hunk 1]
```diff
@@ -238,11 +238,19 @@ int homa_sock_bind(struct homa_socktab *socktab, struct homa_sock *hsk,
 {
 	int result = 0;
 	struct homa_sock *owner;
+	if (port == 0) {
+		// Homa sockets already are bound to "random" ports, so there
+		// is no need to assign a new one. This codepath is needed for
+		// applications that expect to bind a homa port to a specific
```
Comment (danmanj) [inline homa_socktab.c:245]: this should say "homa socket" not "homa port"

**Comment by johnousterhout** [comment]
It's no problem for me to take care of the merge conflicts, but thanks for offering.

-John-

**Comment by johnousterhout** [comment]
I believe that all of these commits have now been merged (with some slight rearrangements and additions, such as unit tests for feature changes). Let me know if I missed anything.

**Pull request closed without merging**